packages/mask/src/plugins/CryptoartAI/SNSAdaptor/ArticleTab.tsx

```diff
@@ -0,0 +1,61 @@
+import type { FC } from 'react'
+import { Link } from '@mui/material'
+import { makeStyles } from '@masknet/theme'
+import { Video } from '../../../components/shared/Video'
+import { CollectibleTab } from './CollectibleTab'
+import { CollectibleState } from '../hooks/useCollectibleState'
+
+const useStyles = makeStyles()({
+    body: {
+        display: 'flex',
+        justifyContent: 'center',
+        height: '300px',
+    },
+    player: {
+        maxWidth: '100%',
+        maxHeight: '100%',
+        border: 'none',
+    },
+})
+
+interface AssetPlayerProps {
+    src?: string
+    alt: string
+}
+
+// opensea supports: JPG, PNG, GIF, SVG, MP4, WEBM, MP3, WAV, OGG, GLB, GLTF.
+const AssetPlayer: FC<AssetPlayerProps> = ({ src, alt }) => {
+    const { classes } = useStyles()
+    if (!src) {
+        return null
+    }
+    const isVideo = src.match(/\.(mp4|webm)$/i)
```

Comment:
https://github.com/DimensionDev/kit/blob/master/src/detect.ts

plz use `detectVideoSupport`

Reply:
see https://caniuse.com/?search=video%20format

Reply:
Still keep using regex so that it will simple and same with https://github.com/DimensionDev/Maskbook/blob/develop/packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx#L29